pkg/cache/cache.go

```diff
@@ -0,0 +1,126 @@
+package cache
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/DanilLagunov/jokes-api/pkg/models"
+)
+
+// Cache struct.
+type Cache struct {
+	sync.RWMutex
+	defaultExpiration time.Duration
+	cleanupInterval   time.Duration
+	items             map[string]Item
+}
+
+// Item struct.
+type Item struct {
+	Value      models.Joke
+	Created    time.Time
+	Expiration int64
+}
+
+// NewCache creating new Cache object.
+func NewCache(defaultExpiration, cleanupInterval time.Duration) *Cache {
+	items := make(map[string]Item)
+
+	cache := Cache{
+		items:             items,
+		defaultExpiration: defaultExpiration,
+		cleanupInterval:   cleanupInterval,
+	}
+
+	if cleanupInterval > 0 {
+		cache.startCleaner()
+	}
+
+	return &cache
+}
+
+// Get return cache item by key.
+func (c *Cache) Get(key string) (models.Joke, error) {
+	c.RLock()
+
+	defer c.RUnlock()
+
+	item, found := c.items[key]
+	if !found {
+		return models.Joke{}, errors.New("key not found")
```

Comment:
when the key is not found, you need to handle it in an appropriate way. It's not an server error